pkg/provider/vault/vault.go

```diff
@@ -715,17 +677,21 @@ func getTypedKey(data map[string]interface{}, key string) ([]byte, error) {
 		return []byte(t), nil
 	case map[string]interface{}:
 		return json.Marshal(t)
+	case []string:
+		return []byte(strings.Join(t, "\n")), nil
 	case []byte:
 		return t, nil
 	// also covers int and float32 due to json.Marshal
 	case float64:
 		return []byte(strconv.FormatFloat(t, 'f', -1, 64)), nil
+	case json.Number:
+		return []byte(t.String()), nil
```

Comment:
One more case is missing found using PKI secret backend:
```suggestion
	case json.Number:
		return []byte(t.String()), nil
	case []interface{}:
		return json.Marshal(t)
```